### ISSUE TITLE
belongsToMany

### ISSUE THREAD

**Issue by thosakwe** [angel-dart/orm#19]

**Comment by sandrobilbeisi** [comment]
I am working on a HABTM implementation for my own ORM
If I have time I'll take a look at Angel ORM and see what it needs to fully support it

**Comment by thosakwe** [comment]
A many-to-many relationship can now be modeled like so.
`UserRole` in this case is a pivot table joining `User` and `Role`.

Note that in this case, the models must reference the private classes (`_User`, etc.), because the canonical versions (`User`, etc.) are not-yet-generated:

```dart
library angel_orm_generator.test.models.user;

import 'package:angel_model/angel_model.dart';
import 'package:angel_orm/angel_orm.dart';
import 'package:angel_serialize/angel_serialize.dart';
import 'package:collection/collection.dart';
part 'user.g.dart';

@serializable
@orm
abstract class _User extends Model {
  String get username;
  String get password;
  String get email;

  @hasMany
  List<_UserRole> get userRoles;

  List<_Role> get roles => userRoles.map((m) => m.role).toList();
}

@serializable
@orm
abstract class _Role extends Model {
  String name;

  @hasMany
  List<_UserRole> get userRoles;

  List<_User> get users => userRoles.map((m) => m.user).toList();
}

@Serializable(autoIdAndDateFields: false)
@orm
abstract class _UserRole {
  int get id;

  @belongsTo
  _User get user;

  @belongsTo
  _Role get role;
}
```

TLDR: 
1. Make a pivot table, C, between two tables, table A and B
2. C should `@belongsTo` both A and B.
3. Both A and B should `@hasMany` C.
4. For convenience, write a simple getter, like the above `User.roles`.

Test: https://raw.githubusercontent.com/angel-dart/orm/master/angel_orm_generator/test/many_to_many_test.dart

There are 2 tests there, but they are more or less a proof-of-concept. All the logic for the other relations have their own unit tests.
